MantidPlot/src/Mantid/MantidPlotUtilities.cpp

```diff
@@ -0,0 +1,71 @@
+#include "MantidPlotUtilities.h"
+
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/Run.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidGeometry/MDGeometry/IMDDimension.h"
+#include <MantidQtMantidWidgets/MantidDisplayBase.h>
+
+using namespace MantidQt::MantidWidgets;
+using Mantid::API::WorkspaceGroup_const_sptr;
+using Mantid::API::WorkspaceGroup_sptr;
+using Mantid::API::MatrixWorkspace_const_sptr;
+using Mantid::API::MatrixWorkspace_sptr;
+using Mantid::API::MatrixWorkspace;
+using Mantid::API::ExperimentInfo;
+using Mantid::HistogramData::Histogram;
+
+/**
+ * Gets the given log value from the given workspace as a double.
+ * Should be a single-valued log!
+ * @param wsIndex :: [input] Index of workspace in group
+ * @param matrixWS :: [input] Workspace to find log from
+ * @param logName :: [input] Name of log
+ * @returns log value as a double, or workspace index
+ * @throws invalid_argument if log is wrong type or not present
+ */
+double getSingleWorkspaceLogValue(
+    size_t wsIndex, const Mantid::API::MatrixWorkspace_const_sptr &matrixWS,
+    const QString &logName) {
+  if (logName == MantidWSIndexWidget::WORKSPACE_INDEX || logName == "") {
+    return wsIndex;
+  } else {
+    // MatrixWorkspace is an ExperimentInfo
+    if (auto ei = boost::dynamic_pointer_cast<const ExperimentInfo>(matrixWS)) {
```

Comment:
`MatrixWorkspace` inherits from `ExperimentInfo`, hence you don't have to perform a `dynamic_pointer_cast` here. You should be able to get the run via `matrixWS->run()`.